### PR TITLE
Rewrite images to kotsadm, include additional images

### PR DESCRIPTION
**Pull request by marccampbell**
This PR moves the functionality of rewriting the application images into the Go kotsadm API while also adding support for additionalImages attribute of the kots.io/v1beta1, Kind=Application manifest.

This is planned for 1.13.0

### CODE REVIEW

**Comment by marccampbell** [comment]
This is a draft for now, we need to finish the rewriteimages function, change the airgap upload to use this, delete the code in typescript, add some tests, and also add additionalImage support in before this is ready

**Comment by marccampbell** [comment]
Remaining todos:

- [x] Push additional images to the local registry
- [ ] The other code paths that rewrite images should be moved over
- [ ] Delete the typescript mutation
- [ ] Handle gitops commits from here
- [ ] The diff calculation needs to be in the go api now

File: api/Dockerfile.skaffold [hunk 1]
```diff
@@ -38,7 +38,7 @@ ADD ./src /src/src
 
 RUN make build
 
-# COPY ./kots.so /lib/kots.so
+COPY ./kots.so /lib/kots.so
```
Comment (divolgin) [inline api/Dockerfile.skaffold]: no!

Comment (marccampbell) [reply api/Dockerfile.skaffold]: I'm excited. for the end state of this when these lines no longer exist

File: pkg/app/registry.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+				logger.Error(err)
+			}
+		}
+	}()
+
+	// get the archive and store it in a temporary location
+	appDir, err := GetAppVersionArchive(a.ID, a.CurrentSequence)
+	if err != nil {
+		finalError = err
+		return errors.Wrap(err, "failed to get app version archive")
```
Comment (divolgin) [inline pkg/app/registry.go]: Don't change this, but I think better pattern is to declare `finalError` in the function declaration and then here:
```
finalError = errors.Wrap(err, "failed to get app version archive")
return
```

**Pull request merged**